runtime-modules/project-token/src/tests/split.rs

```diff
@@ -0,0 +1,807 @@
+    build_default_test_externalities_with_balances(vec![(
+        DEFAULT_ACCOUNT_ID,
+        DEFAULT_SPLIT_ALLOCATION + ExistentialDeposit::get(),
+    )])
+    .execute_with(|| {
+        IssueTokenFixture::default().execute_call().unwrap();
+
+        let result = FinalizeRevenueSplitFixture::default().execute_call();
+
+        assert_err!(result, Error::<Test>::RevenueSplitNotActiveForToken);
+    })
+}
+
+#[test]
+fn finalize_split_ok_with_event_deposit() {
+    build_default_test_externalities_with_balances(vec![(
+        DEFAULT_ACCOUNT_ID,
+        DEFAULT_SPLIT_ALLOCATION + ExistentialDeposit::get(),
+    )])
+    .execute_with(|| {
+        IssueTokenFixture::default().execute_call().unwrap();
+        IssueRevenueSplitFixture::default().execute_call().unwrap();
+        increase_block_number_by(DEFAULT_SPLIT_DURATION);
+
+        FinalizeRevenueSplitFixture::default()
+            .execute_call()
+            .unwrap();
+
+        last_event_eq!(RawEvent::RevenueSplitFinalized(
+            1u64,
+            DEFAULT_ACCOUNT_ID,
+            DEFAULT_SPLIT_ALLOCATION
+        ))
+    })
+}
+
+#[test]
+fn finalize_split_ok_with_token_status_set_to_inactive() {
+    build_default_test_externalities_with_balances(vec![(
+        DEFAULT_ACCOUNT_ID,
+        DEFAULT_SPLIT_ALLOCATION + ExistentialDeposit::get(),
+    )])
+    .execute_with(|| {
+        IssueTokenFixture::default().execute_call().unwrap();
+        IssueRevenueSplitFixture::default().execute_call().unwrap();
+        increase_block_number_by(DEFAULT_SPLIT_DURATION);
+
+        FinalizeRevenueSplitFixture::default()
+            .execute_call()
+            .unwrap();
+
+        assert!(matches!(
+            Token::token_info_by_id(1u64).revenue_split,
+            RevenueSplitState::Inactive,
+        ));
+    })
+}
+
+#[test]
+fn finalize_split_ok_with_leftover_joys_transferred_to_account() {
+    build_default_test_externalities_with_balances(vec![(
+        DEFAULT_ACCOUNT_ID,
+        DEFAULT_SPLIT_ALLOCATION + ExistentialDeposit::get(),
+    )])
+    .execute_with(|| {
+        let treasury_account = treasury_account_for(1u64);
+        IssueTokenFixture::default().execute_call().unwrap();
+        IssueRevenueSplitFixture::default().execute_call().unwrap();
+        increase_block_number_by(DEFAULT_SPLIT_DURATION);
+
+        FinalizeRevenueSplitFixture::default()
+            .execute_call()
+            .unwrap();
+
+        // treasury account final balance == Existential deposit
+        assert_eq!(
+            Joy::<Test>::usable_balance(treasury_account),
+            ExistentialDeposit::get()
+        );
+        // account id balance increased by leftover joy amount
+        assert_eq!(
+            Joy::<Test>::usable_balance(DEFAULT_ACCOUNT_ID),
+            DEFAULT_SPLIT_ALLOCATION + ExistentialDeposit::get()
+        );
+    })
+}
+
+#[test]
+fn participate_to_split_failst_with_invalid_token_id() {
```

Comment:
```suggestion
fn participate_in_split_fails_with_invalid_token_id() {
```
And in general `participate_to_split` => `participate_in_split`